### ISSUE TITLE
Question: url to stream in browser

### ISSUE THREAD

**Issue by mjw56** [mafintosh/peerflix#211]
Hi,

I'd like to possibly use the url that is emitted from peerflix to stream video in the browser. I'm wondering what the best way to do this might be? Also, do I need to always have the vlc flag on in order to get the url?

Thanks.

**Comment by tomekmarchi** [comment]
Check out peerflix-server it's basically already done in that package.

**Comment by jaruba** [comment]
I made an NPAPI Plugin for this:
https://github.com/jaruba/PowderNPAPI

Not sure how useful it will still be considering that NPAPI's will be deprecated in Chrome from September, but I think it's worth checking out. :)

**Comment by tomekmarchi** [comment]
If you want to see how to use peerflix-server, peep this in action. https://github.com/tomekmarchi/Mixee

**Comment by mjw56** [comment]
:+1: Thanks guys, these suggestions are great and exactly what I'm looking to do. I'm going to fiddle more with the peerflix-server and see if I can get it to work with an app I'm working on. Cheers :)

**Comment by jaruba** [comment]
If your working on an app, then the best way to do it is by using websockets to pass the link to the browser page.

**Comment by mjw56** [comment]
@jaruba Yeah, I guess that was more my question as I think I was doing it in a very naive way. Here is how I am currently trying to do it:

```js
app.get('/torrent-stream/:magnet?', function(req, res) {
  var cmd = exec('peerflix "' + req.query.magnet + '" --vlc', function (error, stdout, stderr) {
    res.send({ data: stdout })
  });
});
```

As you can see, I pass magnet link to server which then executes the peerflix command programmatically. This poses two problems:

1) A vlc instance is started automatically and I really want to just stream the video through the browser
2) The ip address is nested inside that stdout and I need to do work to strip it out of there. 

Basically my question is, is there a better way to use peerflix to get this address without running an instance of vlc?

**Comment by asapach** [comment]
There's the `webplay` option that will open the browser for you: https://github.com/mafintosh/peerflix/blob/master/app.js#L36

**Comment by mjw56** [comment]
cool, thanks @asapach! That is def getting closer to what I'm looking for.

**Comment by jaruba** [comment]
@mjw56 Check out [this stackoverflow example](http://stackoverflow.com/questions/9914816/what-is-an-example-of-the-simplest-possible-socket-io-example#answer-9916153) of how to use websockets to send data from an app to a web page.

Is there a particular reason why you are calling ``peerflix`` like that? You can get the peerflix http server link by requiring peerflix directly in any node.js environment.

    var peerflix = require('peerflix'),
        engine = peerflix(magnetLink),
    
    engine.server.on('listening', function() {
        var myLink = 'http://localhost:' + engine.server.address().port + '/';
    });

Where ``magnetLink`` is any magnet link or link/path to a torrent file, and ``myLink`` is the streaming link.

**Comment by mjw56** [comment]
@jaruba ah, sweet success. This is exactly what I am looking for. 

I wasn't aware you could call peerflix with magnet link like that, so that's why I was doing it the naive way. This is the simple solution that I was looking for! I got something working using this with websockets, many thanks @jaruba.

**Comment by jaruba** [comment]
Glad to help, this is the best solution because you could later on send more relevant data (like download %, number of peers, etc.), you could even make your app into a daemon/windows service by using a the [Forever module](https://www.npmjs.com/package/forever). :)

You should also check out the [torrent-stream Readme.md](https://github.com/mafintosh/torrent-stream/blob/master/README.md) that will show all events, etc. that you can use for ``engine``. The only difference will be that in the docs you will use ``peerflix()`` instead of ``torrentStream()`` to initialize the module.

**Comment by tomekmarchi** [comment]
I can second that notion I use this method except with express and socket.io the link I supplied has a ton of code already with all that setup feel free to use as you see fit. Great examples @jaruba ! Websocket method is by far the most effective.
